### ISSUE TITLE
Workflow fails with bind failure to ports on Windows

### ISSUE THREAD

**Issue by ohsayan** [actions/virtual-environments#3275]
**Description**  
Well, I've been having a lot of trouble with workflows failing on actions. This isn't new, but has been an issue ever since we started using actions. Our test suite binds a server process to random ports within the dynamic port range [49152, 65535]. However, just on Windows runners, we get bind failures to multiple ports like ports:
- 49722 ([see this](https://github.com/skytable/skytable/runs/2447647491?check_suite_focus=true#step:12:290)) 
- 49689 ([see this](https://github.com/skytable/skytable/runs/2444951599?check_suite_focus=true#step:11:249))
- 49786 ([see this](https://github.com/skytable/skytable/runs/2419645136?check_suite_focus=true#step:11:249))
- 49714 ([see this](https://github.com/skytable/skytable/runs/2419654414?check_suite_focus=true#step:11:249))
....and a multitude of others.

I suspect that this is caused by GHA using Hyper-V or something, but this is becoming extremely annoying and ridiculous at this point. We've had to re-run several workflows countless times over.

**Question, Bug, or Feature?**:  
Bug

**Virtual environments affected**  
- [x] Windows Server 2019

**Image version**

Image version where you are experiencing the issue.

**Expected behavior**  
Bind to any of the ports in the dynamic port range without any issues

**Actual behavior**  
Fails to bind with `code: 10013, kind: PermissionDenied, message: "An attempt was made to access a socket in a way forbidden by its access permissions."`

**Repro steps**  
Try using any application that attempts to bind to any of the mentioned ports

**Comment by LeonidLapshin** [comment]
Hey, @ohsayan !
We will investigate this issue and return with details soon :)

**Comment by ohsayan** [comment]
Yet [another build failure](https://github.com/skytable/skytable/runs/2447877239#step:12:289).  Port 49718 this time

**Comment by miketimofeev** [comment]
@ohsayan it seems these ports are reserved by windows. `netsh interface ipv4 show excludedportrange protocol=tcp` gives the following results
```
Start Port    End Port      
----------    --------      
        80          80      
      5986        5986      
     47001       47001      
     49672       49771      
     49872       49971
```
Is it possible to avoid these ranges?

**Comment by ohsayan** [comment]
@miketimofeev I think it's possible for us to avoid these ranges. These are the only ones...right? (I'll have to patch the test macro, so just confirming)

**Comment by miketimofeev** [comment]
@ohsayan I've updated the comment a bit, please check it one more time. Yes, these are all at the moment, however, if you can parse the output from `netsh interface ipv4 show excludedportrange protocol=tcp` it would be even safer.

**Comment by ohsayan** [comment]
Great; will close the issue once everything's good. I think you should document this somewhere

**Comment by miketimofeev** [comment]
@ohsayan there was a discussion around it but the final decision was against documenting the ports
https://github.com/actions/virtual-environments/issues/2820#issuecomment-810976520

**Comment by ohsayan** [comment]
Commit skytable/skytable@0673192d5120f9f31a36eb1184347eb62fe031c6 fixes it. Closed!

**Comment by eyuplojistik1** [comment]
pardon bu site nasıl üyr veya abone olunuyor

________________________________________
Gönderen: Sayan Nandan ***@***.***>
Gönderildi: 29 Nisan 2021 Perşembe 07:56
Kime: actions/virtual-environments
Bilgi: Subscribed
Konu: Re: [actions/virtual-environments] Workflow fails with bind failure to ports on Windows (#3275)

Closed #3275<https://github.com/actions/virtual-environments/issues/3275>.

**Comment by eyuplojistik1** [comment]
olmadığım halde yüzlerce mail geliyor bana ??

________________________________________
Gönderen: Sayan Nandan ***@***.***>
Gönderildi: 29 Nisan 2021 Perşembe 07:56
Kime: actions/virtual-environments
Bilgi: Subscribed
Konu: Re: [actions/virtual-environments] Workflow fails with bind failure to ports on Windows (#3275)

Closed #3275<https://github.com/actions/virtual-environments/issues/3275>.

**Comment by eyuplojistik1** [comment]
yardımcı olursanız sevinirim

________________________________________
Gönderen: Sayan Nandan ***@***.***>
Gönderildi: 29 Nisan 2021 Perşembe 07:56
Kime: actions/virtual-environments
Bilgi: Subscribed
Konu: Re: [actions/virtual-environments] Workflow fails with bind failure to ports on Windows (#3275)

Closed #3275<https://github.com/actions/virtual-environments/issues/3275>.
